packages/gatsby/content/advanced/migration.md

```diff
@@ -4,11 +4,31 @@ path: /advanced/migration
 title: "Migration"
 ---
 
+- [General Advice](#general-advice)
+  - [Upgrade to Node 10 or 12](#upgrade-to-node-10-or-12)
+  - [Run the doctor](#run-the-doctor)
+  - [Use `yarn dlx` instead of `yarn global`](#use-yarn-dlx-instead-of-yarn-global)
+  - [Enable the PnP plugin when using Webpack 4](#enable-the-pnp-plugin-when-using-webpack-4)
+  - [Call your scripts through `yarn node` rather than `node`](#call-your-scripts-through-yarn-node-rather-than-node)
+  - [Explicitly call the `pre` and `post` scripts](#explicitly-call-the-pre-and-post-scripts)
+  - [Setup your IDE for PnP support](#setup-your-ide-for-pnp-support)
+  - [Update your configuration to the new settings](#update-your-configuration-to-the-new-settings)
+  - [Don't use `.npmrc` files](#dont-use-npmrc-files)
+  - [Take a look at our end-to-end tests](#take-a-look-at-our-end-to-end-tests)
+  - [Don't use `bundleDependencies`](#dont-use-bundledependencies)
+  - [If required: enable the `node-modules` plugin](#if-required-enable-the-node-modules-plugin)
+- [Troubleshooting](#troubleshooting)
+  - [`Cannot find module [...]`](#cannot-find-module-)
+    - [Make sure you use `resolve@1.9+`](#make-sure-you-use-resolve19)
+  - [`A package is trying to access another package [...]`](#a-package-is-trying-to-access-another-package-)
```

Comment:
Great idea! Do you know if there's a Gatsby plugin that would avoid us from having to synchronize this list? We use another table of content in the FaQ, could make sense to factorize that (we can do it in a separate PR).

Reply:
I have found [`gatsby-remark-table-of-contents`](https://www.gatsbyjs.org/packages/gatsby-remark-table-of-contents/). I will implement it in another PR.